interfaces/builtin/vulkan_driver_libs.go

```diff
@@ -0,0 +1,325 @@
+		&compatibility.CompatSpec{Dimensions: []compatibility.CompatDimension{
+			{Tag: "vulkan", Values: []compatibility.CompatRange{{Min: 1, Max: 1}, {Min: 0, Max: 10}}},
+			{Tag: "ubuntu", Values: []compatibility.CompatRange{{Min: 0, Max: math.MaxUint}}},
+		}}); err != nil {
+		return err
+	}
+
+	// Validate *-source directories
+	for _, sda := range []sourceDirAttr{
+		{attrName: "library-source", isOptional: false},
+		{attrName: "icd-source", isOptional: false},
+		{attrName: "implicit-layer-source", isOptional: true},
+		{attrName: "explicit-layer-source", isOptional: true},
+	} {
+		if err := validateSourceDirs(slot, sda); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (iface *vulkanDriverLibsInterface) LdconfigConnectedPlug(spec *ldconfig.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	// The plug can only be the system plug for the time being
+	return addLdconfigLibDirs(spec, slot)
+}
+
+var _ = symlinks.ConnectedPlugCallback(&vulkanDriverLibsInterface{})
+var _ = interfaces.ConfigfilesUser(&vulkanDriverLibsInterface{})
+
+const (
+	vulkanDriverLibs        = "vulkan-driver-libs"
+	vulkanIcdPath           = "/etc/vulkan/icd.d"
+	vulkanExplicitLayerPath = "/etc/vulkan/explicit_layer.d"
+	vulkanImplicitLayerPath = "/etc/vulkan/implicit_layer.d"
+)
+
+func (iface *vulkanDriverLibsInterface) TrackedDirectories() []string {
+	return []string{vulkanIcdPath, vulkanExplicitLayerPath, vulkanImplicitLayerPath}
+}
+
+func checkVulkanIcdFile(slot *interfaces.ConnectedSlot, icdContent []byte) error {
+	var icdJson struct {
+		Icd struct {
+			ApiVersion  string `json:"api_version"`
+			LibraryPath string `json:"library_path"`
+		} `json:"ICD"`
+	}
+	err := json.Unmarshal(icdContent, &icdJson)
+	if err != nil {
+		return fmt.Errorf("while unmarshalling: %w", err)
+	}
+
+	if icdJson.Icd.ApiVersion == "" {
+		return errors.New("no api_version value found")
+	}
+	if err := checkVulkanApiVersion(slot, icdJson.Icd.ApiVersion); err != nil {
+		return err
+	}
+
+	if icdJson.Icd.LibraryPath == "" {
+		return errors.New("no library_path value found")
+	}
+	// Here we are implicitly limiting library_path to be a file
+	// name instead of a full path.
+	_, err = filePathInLibDirs(slot, icdJson.Icd.LibraryPath)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func checkVulkanApiVersion(slot *interfaces.ConnectedSlot, apiVersion string) error {
+	// Build a compatibility label for this API version and check consistency
+	// with the one specified by the interface.
+	version := strings.Split(apiVersion, ".")
+	if len(version) < 2 {
+		return fmt.Errorf("api_version is not a version: %s", apiVersion)
+	}
+	major, err := strconv.Atoi(version[0])
+	if err != nil {
+		return fmt.Errorf("while parsing major: api_version %s", apiVersion)
+	}
+	minor, err := strconv.Atoi(version[1])
+	if err != nil {
+		return fmt.Errorf("while parsing minor: api_version %s", apiVersion)
+	}
```

Comment:
more boring variant (since we're taking all inputs), but also we're not expecting negative numbers, are we?

```suggestion
	var major, minor uint
	if _, err := fmt.Sscanf(apiVersion, "%u.%u", &major, &minor); err != nil {
		return fmt.Errorf("cannot parse api_version: %w", err)
	}
```